File: problem4/problem.js
function largestPalindromeProduct(digits) {
    let palindromes = []
    let n = ''
    while (n.length < digits) {
        n += '9'
    }
    for (let x = parseInt(n); x > 0; x--) {
        for (let y = parseInt(n); y > 0; y--) {
            let f = x*y
            let s = f.toString()
            let p1 = s.substring(0, s.length/2)
            let p2 = s.substring(s.length/2, s.length).split("").reverse().join("")
            if (p1 == p2) {
                palindromes.push(parseInt(f))
            } 
        }
    }
    return Math.max.apply(null,palindromes)
}

console.log(largestPalindromeProduct(2))
console.log(largestPalindromeProduct(3))

File: problem4/problem.py
def largestPalindromeProduct(digits):
    palindromes = []
    n = ''
    while len(n) < digits:
        n += '9'
    n = int(n)
    for x in reversed(range(0,n+1)):
        for y in reversed(range(0,n+1)):
            f = x*y
            s = str(f)
            p1 = s[:len(s)//2]
            p2 = s[len(s)//2:]
            if p1 == p2[len(p2)::-1]:
                palindromes.append(f)
    return max(palindromes)